Sources/Fluent/Query/QueryRepresentable.swift

```diff
@@ -173,10 +173,18 @@ extension QueryRepresentable where Self: ExecutorRepresentable {
 extension QueryRepresentable where Self: ExecutorRepresentable {
     /// Attempts to delete all entities
     /// in the model's collection.
-    public func delete() throws {
+    public func delete(shouldForceDelete: Bool = true) throws {
```

Comment:
Instead of adding `shouldForceDelete: Bool = true` I think you can add an extension where the entity type is soft deletable. Something like:

```swift
extension QueryRepresentable where Self: ExecutorRepresentable, E: SoftDeletable {
    public func forceDelete() { ... }
}
```

Similar to what is done here: https://github.com/vapor/fluent/blob/89bcea79fa31c7bb7ab507e9e455034f68e56ceb/Sources/Fluent/Entity/SoftDeletable.swift#L87-L94